clap_builder/src/builder/styled_str.rs

```diff
@@ -29,6 +31,11 @@ impl StyledStr {
         Self(String::new())
     }
 
```

Comment:
This new interpolate function replaces the text lookup keys with the actual texts. There is probably a better place to put this, but for the purposes of demonstrating the approach, I thought this was good enough.